### PR TITLE
cpp: clarify object destruction on deletion

### PR DESCRIPTION
**Pull request by tomaszkapela**
There is a slight but significant difference in atomic/transactional
deallocation that needs to be clarified.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmem.github.io/40)

### CODE REVIEW

**Comment by krzycz** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[_posts/2016-05-19-cpp-06.md, line 140 \[r1\]](https://reviewable.io:443/reviews/pmem/pmem.github.io/40#-KIXd6JhrC4no0npBMOe:-KIXd6JhrC4no0npBMOf:-587460367) ([raw file](https://github.com/pmem/pmem.github.io/blob/bebc44ce737e1aafb495e018df1b04a0ead4739b/_posts/2016-05-19-cpp-06.md#L140)):*
> ```Markdown
> transactional versions can only be used within transactions. If used outside of
> transaction scope, an exception is thrown. In case of _atomic_ deallocations the
> memory gets freed, but the object's destructor is **never called**.
> ```

I would rephrase the entire paragraph. For me, it's still somewhat unclear.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmem.github.io/40)*

**Comment by tomaszkapela** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[_posts/2016-05-19-cpp-06.md, line 140 \[r1\]](https://reviewable.io:443/reviews/pmem/pmem.github.io/40#-KIXd6JhrC4no0npBMOe:-KIXvHkDmjr8wYhEQmjH:-1743137966) ([raw file](https://github.com/pmem/pmem.github.io/blob/bebc44ce737e1aafb495e018df1b04a0ead4739b/_posts/2016-05-19-cpp-06.md#L140)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary>
> I would rephrase the entire paragraph. For me, it's still somewhat unclear.
</details>
So I used the best method known to engineers, examples. HTH

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmem.github.io/40)*

**Comment by pbalcer** [comment]
:lgtm:

<details><summary><i>Previously, tomaszkapela wrote…</i></summary>
> #### cpp: clarify object destruction on deletion
> 
> There is a slight but significant difference in atomic/transactional
> 
> deallocation that needs to be clarified.
> 
> 
> 
</details>


---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmem.github.io/40#-:-KJ0zcyIvNW-_eQnu9bj:1417088721)*

**Comment by pbalcer** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmem.github.io/40)*
